silq/analysis/analysis.py

```diff
@@ -715,8 +717,44 @@ def analyse_EPR(empty_traces: np.ndarray,
             'mean_high_blip_duration': results_read['mean_high_blip_duration']}
 
 
-def analyse_flips(up_proportions_arrs: List[np.ndarray],
-                  threshold_up_proportion: Union[Sequence, float]):
+def analyse_threshold_up_proportion(up_proportions_arrs: np.ndarray,
+                                    shots_per_frequency: int):
+    if shots_per_frequency < 25:
+        num_shots = 25   # to make sure algorithm works
```

Comment:
This comment is quite unclear, why 25? Is it that it otherwise has too few points to converge? The reason should be clearly explained in a comment, especially because it seems like an arbitrary value.
Also these four lines can be replaced by: `num_shots = max(shots_per_frequency, 25)